superduperdb/ext/transformers/checkpoint.py

```diff
@@ -0,0 +1,21 @@
+import typing as t
+
+from superduperdb.components.component import Component
+from superduperdb.components.datatype import DataType, file_lazy
+
+
+class Checkpoint(Component):
+    """Checkpoint component for saving the model checkpoint.
+
+    :param path: The path to the checkpoint.
+    :param step: The step of the checkpoint.
+    """
+
+    path: t.Optional[str]
+    step: int
+    _artifacts: t.ClassVar[t.Sequence[t.Tuple[str, DataType]]] = (("path", file_lazy),)
+    type_id: t.ClassVar[str] = "checkpoint"
```

Comment:
Shouldn't this be in `components/*.py`?

Reply:
Currently, it is only used in transformers, so I put it in transformers first. If it will be used in more places in the future, then it can be moved to a higher level like `components/*.py`.

WDYT?